src/tools/fuzzing/fuzzing.cpp

```diff
@@ -2599,7 +2602,31 @@ Expression* TranslateToFuzzReader::makeBasicRef(Type type) {
       return null;
     }
     case HeapType::string: {
-      return makeString();
+      // In non-function contexts all we can do is string.const.
+      if (!funcContext) {
+        return makeStringConst();
+      }
+      switch (upTo(9)) {
+        case 0:
+        case 1:
+        case 2:
+          return makeStringConst();
+        case 3:
+        case 4:
+        case 5:
+          return makeStringNewCodePoint();
+        case 6:
+        case 7:
+          // We less frequently make string.new_array as it may generate a lot
+          // of code for the array in some cases.
+          return makeStringNewArray();
+        case 8:
+          // We much less frequently make string.concat as it will recursively
+          // generate two string children, i.e., it can lead to exponential
+          // growth.
```

Comment:
Isn't this true of lots of instructions, though? Don't we have some depth limiting mechanism that already takes care of this?

Reply:
We do have a limit, yeah, but it is nice to avoid reaching it I think (as reaching it has downsides - we've exhausted input bytes). And this still gives a reasonable chance for concat to happen I think.